lib/mail/version_specific/ruby_1_9.rb

```diff
@@ -125,7 +125,11 @@ def Ruby19.q_value_decode(str)
     def Ruby19.param_decode(str, encoding)
       str = uri_parser.unescape(str)
       str = charset_encoder.encode(str, encoding) if encoding
-      str
+      decoded = str.encode(Encoding::UTF_8, :invalid => :replace, :replace => "")
+      decoded.valid_encoding? ? decoded : decoded.encode(Encoding::UTF_16LE, :invalid => :replace, :replace => "").encode(Encoding::UTF_8)
```

Comment:
In ruby 1.9.3 "utf-8" to "utf-8" encoding is a no-op:

```ruby
[ ruby-1.9.3-p547 ][ mail:007:0 ] >> invalid_char = "\x8A"
=> "\x8A"
[ ruby-1.9.3-p547 ][ mail:008:0 ] >> invalid_char.encoding
=> #<Encoding:UTF-8>
[ ruby-1.9.3-p547 ][ mail:009:0 ] >> invalid_char.valid_encoding?
=> false
[ ruby-1.9.3-p547 ][ mail:012:0 ] >> converted = invalid_char.encode(Encoding::UTF_8, :undef => :replace, :invalid => :replace, :replace => "")
=> "\x8A"
[ ruby-1.9.3-p547 ][ mail:013:0 ] >> converted.valid_encoding?
=> false
```

The conversion to utf_16 and back is a workaround for that no-op